internal/services/keyvault/key_vault_certificate_resource.go

```diff
@@ -902,10 +926,35 @@ func expandKeyVaultCertificatePolicy(d *pluginsdk.ResourceData) (*keyvault.Certi
 
 	issuers := policyRaw["issuer_parameters"].([]interface{})
 	issuer := issuers[0].(map[string]interface{})
-	policy.IssuerParameters = &keyvault.IssuerParameters{
+	issuerName := issuer["name"].(string)
+
+	certificateType := ""
+	if raw, ok := issuer["certificate_type"]; ok && raw != nil {
+		if v, ok := raw.(string); ok {
+			certificateType = strings.TrimSpace(v)
+		}
+	}
+
+	if certificateType != "" {
+		if !strings.EqualFold(issuerName, "DigiCert") {
+			return nil, fmt.Errorf("`certificate_type` can only be specified when the issuer is DigiCert")
+		}
+
+		if !isValidDigiCertCertificateType(certificateType) {
+			return nil, fmt.Errorf("`certificate_type` must be one of [%s] when the issuer is DigiCert", strings.Join(validDigiCertCertificateTypes, ", "))
+		}
```

Comment:
It's too late to do this validation inside expand / create function -- this logic executes during apply time. Please move these logics into the CustomizeDiff function so the validation happen during plan time.

Search for the string `CustomizeDiff: func(` to find examples how it's implemented in other resources.